vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -371,7 +372,7 @@ def apply(
 
         return w8a8_scaled_mm_func(qinput=qinput,
                                    weight=weight,
-                                   out_dtype=input.dtype,
+                                   out_dtype=out_dtype,
```

Comment:
When do these two types diverge? It looks like this is generally just an FP8 input, FP8 output layer? Were you seeing some problem as a result of this not being set correctly or did you just notice it while you were here?

Reply:
The divergence was added by #14578.

In normal case, this layer layer should be bf16/fp16 in and bf16/fp16 out. But when we have fp8 input, we still want bf16/fp16 out.